packages/algorand-js/test/src/interpreter/opcode-list.ts

```diff
@@ -412,4 +413,187 @@ describe("Teal Opcodes", function () {
+        () => op.execute(stack),
+        ERRORS.TEAL.ZERO_DIV
+      );
+    });
+  });
+
+  describe("Bitwise OR", function () {
+    const stack = new Stack<StackElem>();
+
+    it("should return correct bitwise-or of two unit64", function () {
+      stack.push(BigInt("10"));
+      stack.push(BigInt("20"));
+      const op = new BitwiseOr();
+      op.execute(stack);
+
+      const top = stack.pop();
+      assert.equal(top, BigInt("30"));
+    });
+
+    it("should throw error with bitwise-or if stack is below min length", function () {
+      stack.push(BigInt("1")); // stack.length() = 1
+      const op = new BitwiseOr();
+      expectTealError(
+        () => op.execute(stack),
+        ERRORS.TEAL.ASSERT_STACK_LENGTH
+      );
+    });
+
+    it("should throw error if bitwise-or is used with strings", function () {
+      stack.push(toBytes("str1"));
+      stack.push(toBytes("str2"));
+      const op = new BitwiseOr();
+      expectTealError(
+        () => op.execute(stack),
+        ERRORS.TEAL.INVALID_TYPE
+      );
+    });
+  });
+
+  describe("Bitwise AND", function () {
+    const stack = new Stack<StackElem>();
+
+    it("should return correct bitwise-and of two unit64", function () {
+      stack.push(BigInt("10"));
+      stack.push(BigInt("20"));
+      const op = new BitwiseAnd();
+      op.execute(stack);
+
+      const top = stack.pop();
+      assert.equal(top, BigInt("0"));
+    });
+
+    it("should throw error with bitwise-and if stack is below min length", function () {
+      stack.push(BigInt("1")); // stack.length() = 1
+      const op = new Add();
+      expectTealError(
+        () => op.execute(stack),
+        ERRORS.TEAL.ASSERT_STACK_LENGTH
+      );
+    });
+
+    it("should throw error if bitwise-and is used with strings", function () {
+      stack.push(toBytes("str1"));
+      stack.push(toBytes("str2"));
+      const op = new BitwiseAnd();
+      expectTealError(
+        () => op.execute(stack),
+        ERRORS.TEAL.INVALID_TYPE
+      );
+    });
+  });
+
+  describe("Bitwise XOR", function () {
+    const stack = new Stack<StackElem>();
+
+    it("should return correct bitwise-xor of two unit64", function () {
+      stack.push(BigInt("10"));
+      stack.push(BigInt("20"));
+      const op = new BitwiseXor();
+      op.execute(stack);
+
+      const top = stack.pop();
+      assert.equal(top, BigInt("30"));
+    });
+
+    it("should throw error with bitwise-xor if stack is below min length", function () {
+      stack.push(BigInt("1")); // stack.length() = 1
+      const op = new BitwiseXor();
+      expectTealError(
+        () => op.execute(stack),
+        ERRORS.TEAL.ASSERT_STACK_LENGTH
+      );
+    });
+
+    it("should throw error if bitwise-xor is used with strings", function () {
+      stack.push(toBytes("str1"));
+      stack.push(toBytes("str2"));
+      const op = new BitwiseXor();
+      expectTealError(
+        () => op.execute(stack),
+        ERRORS.TEAL.INVALID_TYPE
+      );
```

Comment:
for checking failure on strings, you can have a function and use it in tests multiple times:
```
function execExpectError(stack, strs: string[],op: Op, err) {
  return () => {
    for (let s of strs) stack.push(s)
    expectTealError(() => op.execute(stack), err)
  }
}

it("should ...", execExpectError("str1", "str2", newe BitwiseXor(), ERRORS.TEAL.INVALID_TYPE)
```

And you can reuse it in few places above.

Reply:
Yes. Thanks